modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/AtomicFieldsLengthValidator.scala

```diff
@@ -146,10 +146,16 @@ object AtomicFieldsLengthValidator {
         Monad[F].pure(Right(()))
     }
 
-  private def validateField(event: EnrichedEvent)(atomicField: AtomicField): ValidatedNel[String, Unit] = {
+  private def validateField(
+    event: EnrichedEvent,
+    atomicFieldLimits: Map[String, Int]
+  )(
+    atomicField: AtomicField
+  ): ValidatedNel[String, Unit] = {
     val actualValue = atomicField.enrichedValueExtractor(event)
-    if (actualValue != null && actualValue.length > atomicField.maxLength)
-      s"Field ${atomicField.name} longer than maximum allowed size ${atomicField.maxLength}".invalidNel
+    val maxLength = atomicFieldLimits.getOrElse(atomicField.name, Int.MaxValue)
```

Comment:
I think this line is a problem.  It does a look-up by string name for **every** field name (124 fields) and for **every** event (~1000 events per second).

We won't know the performance impact of that until we measure it.  But I'm not hopeful.

I think it can be avoided..... if you leave the `AtomicField` class definition as it was before:

```
case class AtomicField(
    name: String,
    enrichedValueExtractor: EnrichedEvent => String,
    maxLength: Int
  )
```

...and then build your `List[AtomicField]` once when the app first starts up.  To build the list, you need to do the expensive(ish) look-up of lengths from the Map.  But once you've built the `List[AtomicField]` then it will be cheap to validate each event.